tests/base/src/test/java/org/wildfly/security/auth/TestLoginModule.java

```diff
@@ -82,11 +80,14 @@ public boolean login() throws LoginException {
 
     @Override
     public boolean commit() throws LoginException {
-        this.subject.getPrincipals().add(this.principal);
-        // add a caller principal group for testing purposes.
-        final Group group = new TestGroup("CallerPrincipal");
-        group.addMember(new NamePrincipal("auth-caller"));
-        this.subject.getPrincipals().add(group);
+        this.subject.getPrincipals().add(new NamePrincipal("whoami"));
+        this.subject.getPrincipals().add(new AnonymousPrincipal());
+        this.subject.getPrincipals().add(new Roles("Admin"));
+        this.subject.getPrincipals().add( new Roles("User"));
+        this.subject.getPrincipals().add(new Roles("Guest"));
```

Comment:
+1 on this being documented, I think the use if plural / singular attribute names will just be a side effect of the mapping.  Otherwise we could come up with a more complex mapping configuration but I don't think we would benefit from that really most importantly the attribute names are predictable.